### ISSUE TITLE
Make variable order for dict keys configurable

### ISSUE THREAD

**Issue by cuteufo** [microsoft/debugpy#317]
I searched 'sort variables' but didn't see any relevant issues. If I missed something, please forgive and let me know. Thanks.

Can we have a feature of 'sort variables' in debugger? this is very handy when dealing with big chunk of variables and need to do cross check among the variables. PyCharm has this feature and I use it as a default setting.

Edit: More specifically, by 'variables', I mean the keys in a dictionary. Check the attached video, you will see what I mean.
[sortvariables.mp4.zip](https://github.com/microsoft/debugpy/files/4839695/sortvariables.mp4.zip)

**Comment by cuteufo** [comment]
Thanks. Please refer to my edited OP.

**Comment by cuteufo** [comment]
Terribly sorry, I didn't realize I was in a wrong repository. I closed this issue. Hope that doesn't bother you too much.

**Comment by int19h** [comment]
This is the correct repository for your issue - vscode-python is the overall extension, and debugpy is the debugger that it uses. This is a debugger feature, hence why I moved it here.

After watching the video, I see that you mean dict keys specifically (if you experiment with local or member variables, you'll see that they are sorted). Now, for dicts, the reason why we preserve the original order is because Python itself does so since 3.6, and  thus Python code may rely on that order - since it's potentially significant, we keep it so as not to lose information.

**Comment by cuteufo** [comment]
I don't quite agree with this. How Python code rely on the order of dictionary keys is different how I need them to be shown in a debugger pane -- imagine I have a big dictionary with more than 30 key-value pairs and I need to monitor multiple keys during the debugging process.

**Comment by int19h** [comment]
Thing is, some code might not care about it, and other code does - and we have no way of knowing. A lot of metaprogramming code that's written for 3.7+ relies on ordered `kwargs` and class definition dicts (in metaclasses). So if we always sort it, it loses important information that somebody might need.

It would be nice to have this as an option, but ideally it would go somewhere into the context menu on the Variables pane, so that you can easily check & uncheck it. That would require an extensibility point on VSCode side of things.

For now, the best way to do what you want is to utilize Watch - you can cherry-pick the keys that you want (it's okay if some of them are missing when you add a watch), and have them in the exact order that you need to monitor.

**Comment by cuteufo** [comment]
Thanks, to be clear, yes, I’m looking for this option to sort the keys SHOWING in the debugger pane, not necessarily to change its order in Python code, exactly what Pycharm does.

**Comment by int19h** [comment]
To clarify - on the debugpy & vscode-python side, the best we could possibly do is provide a debug configuration property for this, just like we do for variable grouping. But then you would have to configure this separately for every config in every launch.json, and you wouldn't be able to switch it dynamically.

A proper implementation of this - with dynamic switching - would require additions to DAP (debug adapter protocol), and support for those in VSCode.

**Comment by sylph520** [comment]
Yes, I agreed. The sorting feature is handy, per launch.json switching configuration is reasonable, any progresses here?
